scripts/oauth_token/oauth_token/authentication.py

```diff
@@ -0,0 +1,150 @@
+import json
+import os
+import time
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, cast
+
+import requests
+
+
+@dataclass
+class AuthInfo:
+    host: str
+    access_token: str
+    member_id: int
+
+
+@dataclass
+class IdentityProviderCredentials:
+    username: str
+    password: str
+
+
+class QI2API:
+    def __init__(self, host: str):
+        self.host = host
+
+    def fetch_team_member_id(self, access_token: str) -> int:
+        members_url = f"{self.host}/members"
+        headers = {"Authorization": f"Bearer {access_token}"}
+
+        response = requests.get(members_url, headers=headers)
+        response.raise_for_status()
+        members_pages = cast(dict[str, Any], response.json())
+        members = members_pages["items"]
+
+        if len(members) == 1:
+            member_id = members[0]["id"]
+            return cast(int, member_id)
+
+        raise ValueError("Authentication associated with more than one team member")
```

Comment:
Maybe an `AssertionError` or `LookupError` would be more suitable here? IIUC ValueError should indicate the argument passed to this method has a wrong value.

Reply:
This error is the same one we have in the original script. I tried not to touch everything, but I will change it